docs/reference/search/search-your-data/search-using-query-rules.asciidoc

```diff
@@ -0,0 +1,150 @@
+[[search-using-query-rules]]
+== Searching with query rules
+++++
+<titleabbrev>Searching with query rules</titleabbrev>
+++++
+
+[[query-rules]]
+
+preview::[]
+
+_Query rules_ are rules that are defined inside {es} that allow customization of search results for queries that match specified criteria metadata.
+This allows for more control over promoted results, ensuring that specified documents always match defined criteria and are returned at the top of the result list.
+Metadata is defined in the query rule, and is matched against the query criteria.
+Metadata is a string key, and can be anything from a user-entered query string, personalized metadata about users (e.g. country, language, etc). or anything else defined in the rules.
+
+Rules are defined using the <<query-rules-apis,Query rules API>> and searched using the <<query-dsl-rule-query,rule query>>.
+
+When defining a rule, consider the following:
+
+* The type of rule we want to apply:
++
+--
+* `pinned` will re-write the query into a <<query-dsl-pinned-query, pinned query>>, pinning specified results matching the query rule at the top of the returned result set.
+--
+* The criteria for which this rule will match. Criteria is defined as `type`, `metadata`, and `values`. The allowed types are:
++
+--
+* `exact` will match when the rule metadata matches the specified value exactly.
+* `fuzzy` will match when the rule metadata matches the specified value within an allowed {wikipedia}/Levenshtein_distance[Levenshtein edit distance].
+* `prefix` will match when the rule metadata starts with the specified value.
+* `suffix` will match when the rule metadata ends with the specified value.
+* `contains` will match when the rule metadata contains the specified value.
+* `lt` will match when the rule metadata is less than the specified value.
+* `lte` will match when the rule metadata is less than or equal to the specified value.
+* `gt` will match when the rule metadata is greater than the specified value.
+* `gte` will match when the rule metadata is greater than or equal to the specified value.
+* `always` will always match for all rule queries.
+--
+* The actions to take when the rule matches a query:
++
+--
+* `ids` will pin the specified <<mapping-id-field,`_id`>>s.
```

Comment:
```suggestion
* `ids` will pin the specified <<mapping-id-field,`_id`>>s from a single index.
```

Reply:
_Technically_ if we have the same ID in two indices then both will be pinned.